ste/xfer-anyToRemote.go

```diff
@@ -141,6 +141,18 @@ func anyToRemote(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer, sen
 		}
 	}
 
+	// step 5a: lock the destination
+	// (is safe to do it relatively early here, before we run the prologue, because its just a internal lock, within the app)
+	// But must be after all of the early returns that are above here (since
+	// if we succeed here, we need to know the epilogue will definitely run to unlock later)
+	err = jptm.WaitUntilLockDestination(jptm.Context())
```

Comment:
>err = jptm.WaitUntilLockDestination(jptm.Context()) [](start = 1, length = 51)

I'm a bit confused here, why are we locking the destination when the destination is the remote?

The two scenarios (url encoding on windows and stripping .zip) we originally wanted to address are both for download, right?

Reply:
But we also do the exact same encoding stuff for uploads to Azure Files, since it has the same naming restrictions as Windows.

Reply:
BTW, your comment made be realise that I'm not setting the case-sensitivity right.  Will need to think about that.

Reply:
OK. Have just pushed changed to correct the bug, it now takes both fromTo and OS into account when setting case sensitivity